sparse/coo.py

```diff
@@ -1228,6 +1228,14 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         else:
             return NotImplemented
 
+    def __array__(self, dtype=None, **kwargs):
+        x = self.todense()
+        if dtype and x.dtype != dtype:
+            x = x.astype(dtype)
+        if not isinstance(x, np.ndarray):
```

Comment:
Maybe [`np.asarray`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.asarray.html)? It passes through things that are already arrays.

But, in any case, do we really need this check, since `.todense()` already returns an `np.ndarray`.

Reply:
Removed